sdk/identity/azure-identity/src/test/java/com/azure/identity/WorkloadIdentityCredentialTest.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.util.Configuration;
+import com.azure.identity.implementation.IdentityClient;
+import com.azure.identity.util.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.MockedConstruction;
+import reactor.test.StepVerifier;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.UUID;
+
+import static org.mockito.Mockito.mockConstruction;
+import static org.mockito.Mockito.when;
+
+public class WorkloadIdentityCredentialTest {
+
+    private static final String CLIENT_ID = UUID.randomUUID().toString();
+
+    @Test
+    public void testWorkloadIdentityFlow() {
+        // setup
+        String endpoint = "https://localhost";
+        String token1 = "token1";
+        TokenRequestContext request1 = new TokenRequestContext().addScopes("https://management.azure.com");
+        OffsetDateTime expiresAt = OffsetDateTime.now(ZoneOffset.UTC).plusHours(1);
+        Configuration configuration = Configuration.getGlobalConfiguration().clone();
+        configuration.put(Configuration.PROPERTY_AZURE_AUTHORITY_HOST, endpoint); // This must stay to signal we are in an app service context
```

Comment:
What's this comment about? Definitely no App Service here

Reply:
that's copy pasta comment.